File: sort/quick sort/src.js
/**
 * @param {Number[]} arr unsorted array
 * @return {Number[]} Sorted array
 */
function quickSort(arr) {

    // base case
    if (arr.length <= 1) {
        return arr;
    }

    // Last element in the array
    const pivotIndex = arr.length - 1;
    const lesser = [];
    const greater = [];

    for (let i = 0, l = arr.length; i < l; i++) {

        if (i === pivotIndex) {
            continue;
        }

        if (arr[i] < arr[pivotIndex]) {
            lesser.push(arr[i]);
        } else {
            greater.push(arr[i]);
        }
    }

    const sortedLesser = quickSort(lesser);
    const sortedGreater = quickSort(greater);
    return [].concat(sortedLesser, arr[pivotIndex], sortedGreater);
}

const sample = [9, 2, 44, 12, 6, 99, 134];
const sorted = quickSort(sample);

console.log('Unsorted: ' + sample);
console.log('Sorted: ' + sorted);

File: sort/quick sort/src.py

def quickSort(arr):

    if len(arr) <= 1:
        return arr

    pivotIndex = len(arr) - 1
    lesser = []
    greater = []

    for index in range(len(arr)):
        if index == pivotIndex:
            continue

        if arr[index] < arr[pivotIndex]:
            lesser.append(arr[index])
        else:
            greater.append(arr[index])

    return quickSort(lesser) + [arr[pivotIndex]] + quickSort(greater)


sample = [5, 2, 9, 3, 8, 1, 66]
sorted = quickSort(sample)
assert sorted == [1, 2, 3, 5, 8, 9, 66]
print("Unsorted:", sample)
print("Sorted:", sorted)
